$MirOS: ports/lang/llvm/patches/patch-tools_clang_lib_Lex_PPMacroExpansion_cpp,v 1.1 2008/11/10 02:18:29 tg Exp $

	Support for 64-bit time_t.

--- tools/clang/lib/Lex/PPMacroExpansion.cpp.orig	Thu Jan  1 01:00:00 1970
+++ tools/clang/lib/Lex/PPMacroExpansion.cpp	Fri Dec  4 21:06:25 2009
@@ -462,8 +462,8 @@ static void ComputeDATE_TIME(SourceLocat
   };
 
   char TmpBuffer[100];
-  sprintf(TmpBuffer, "\"%s %2d %4d\"", Months[TM->tm_mon], TM->tm_mday,
-          TM->tm_year+1900);
+  snprintf(TmpBuffer, sizeof(TmpBuffer), "\"%s %2d %4d\"", Months[TM->tm_mon], TM->tm_mday,
+          (int)(TM->tm_year+1900));
 
   Token TmpTok;
   TmpTok.startToken();
